MineInAbyss/src/main/java/com/derongan/minecraft/mineinabyss/Ascension/AscensionListener.java

```diff
@@ -115,5 +117,10 @@ public void onPlayerDeath(PlayerDeathEvent deathEvent) {
         Layer layerOfDeath = context.getPlayerDataMap().get(player.getUniqueId()).getCurrentLayer();
 
         deathEvent.setDeathMessage(deathEvent.getDeathMessage() + layerOfDeath.getDeathMessage());
+
+        if (player.getGameMode() == GameMode.SURVIVAL || player.getGameMode() == GameMode.ADVENTURE) { //reset thousand men pins
```

Comment:
Why do we only reset it if its adventure or survival? What reason to preserve for spec/creative?

Add a comment about why if you can

Reply:
Typically on the server people in creative or spectator die by accident or to resolve a bug.
I see no reason to remove pins in that case, we won't have real gameplay in those modes and it could annoy testers

Reply:
Ok